core/include/scipp/core/element/rounding.h

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Samuel Jones
+#pragma once
+
+#include <cmath>
+
+#include "scipp/common/index.h"
+
+namespace scipp::core::element {
+
+constexpr auto floor = [](const auto &a) { return std::floor(a); };
```

Comment:
Missing handling for units and variances (this is why you get weird compiler errors). Variance handling can be disable using flag, if required.